File: hackerrank/day10/main.py
#!/bin/python3

import math
import os
import random
import re
import sys



if __name__ == '__main__':
    n = int(input().strip())

    binary = bin(n)[2:]

    max_count = 0
    current_count = 0

    for c in binary:
        if c == '1':
            current_count += 1
            max_count = max(max_count, current_count)
        else:
            current_count = 0

    print(max_count)

File: hackerrank/day10/main.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}



function main() {
    const n = parseInt(readLine().trim(), 10);

    const binary = n.toString(2);

    let maxCount = 0;
    let currentCount = 0;

    for (const c of binary) {
        if (c === '1') {
            currentCount++;
            maxCount = Math.max(maxCount, currentCount);
        } else {
            currentCount = 0;
        }
    }

    console.log(maxCount);
}